google/cloud/internal/opentelemetry_context.h

```diff
@@ -70,8 +81,8 @@ class ScopedOTelContext {
 
   ~ScopedOTelContext() {
     if (same_thread_) return;
-    for (auto const& c : contexts_) {
-      DetachOTelContext(c);
+    for (auto it = contexts_.rbegin(); it != contexts_.rend(); ++it) {
```

Comment:
Seems like this is some kind of "Remove Common Suffix", something like:

```cc
~ScopedOTelContext() {
...
  DetachOTelMany(std::move(context_));
}

void DetachOTelMany(std::vector<opentelemetry::context::Context> c) {
  auto& v = ThreadLocalOTelContext();
  auto p = std::mismatch(c.rbegin(), c.rend(), v.rbegin(), r.end(), [](auto const& a, auto const& b) {
    return a == b.first;
  });
  std::erase(v.rend(), p.second);
}
```

https://en.cppreference.com/w/cpp/algorithm/mismatch

Reply:
TIL: `std::mismatch`

> Seems like this is some kind of "Remove Common Suffix"

That is what I am trying to do. But I am struggling to erase using the reverse iterators.

Reply:
https://godbolt.org/z/raG9K4ocW though I think it does not matter for small collections.